scipy/stats/tests/test_distributions.py

```diff
@@ -999,6 +999,12 @@ def test_ppf_underflow(self):
         # this should not underflow
         assert_allclose(stats.geom.ppf(1e-20, 1e-20), 1.0, atol=1e-14)
 
+    def test_entropy(self):
+        # Simple tests of entropy.
+        h = stats.geom(np.array([0.0146, 0.2, 0.5, 0.8])).entropy()
+        expected_h = np.array([5.21939796, 2.50201212, 1.38629436, 0.62550303])
+        assert_allclose(h, expected_h)
```

Comment:
Let's target the test at the issue. The other cases would already pass in main, and `_check_private_entropy` in `test_discrete_basic` already checks for consistency between `_entropy` and `_pmf`.

```suggestion
    def test_entropy_gh18226(self):
        # gh-18226 reported that `geom.entropy` produced a warning and
        # inaccurate output for small p. Check that this is resolved.
        h = stats.geom(0.0146).entropy()
        assert_allclose(h, 5.219397961962308, rtol=1e-15)
```

Normally, I would include code that generates the reference value, but I will omit that here because I couldn't decide what the reference should be. Here is one possibility.

<details>

```python3
from mpmath import mp
mp.dps = 100
p = mp.mpf(0.0146)
def f(k):
    return p * (1 - p) ** (k - 1)
float(mp.nsum(lambda k: -f(k) * mp.log(f(k)), (1, mp.inf)))
```

</details>